### PR TITLE
Removes Codable from definitions

### PR DESCRIPTION
**Pull request by Matt54**
We should let the consumer of the package add Codable conformances if they need it. Tonic does not depend on these conformances.

The reason being that Swift doesn't allow you to override the default encode/decode behavior if defined in the package. This is inflexible for the consumer of the package.

Example of a problem that I just ran into:
I defined my own coding keys and encode/decode methods for these structs and used them in my save files. I did this because at the time of writing the code our raw values were integers and I wanted keys which would not change with future updates to the package. However, now that we added Codable to the package I am forced to use the package's raw values for coding keys and the default encode/decode. This breaks my save files in a way that would require a significant refactor to resolve.

TLDR - let the developer using the package add this in their codebase:
extension ChordType: Codable {}

### CODE REVIEW

**Review by aure — APPROVED** [review]

**Comment by SebastianBoldt** [comment]
Shouldn't we then remove it from all the other types?

* Chord
* Type
* ChordType
* ChordProgression etc.
* Pitch
* NoteClass
* Note
* Letter
* Interval
* Accidental
